### ISSUE TITLE
[Phase 2] 統合テスト - XOR 問題

### ISSUE THREAD

**Issue by Showhey798** [Showhey798/gradflow#13]
## 概要

Phase 2 の自動微分機能を使って、XOR 問題を解きます。

## テスト項目

- [ ] 2 層ニューラルネットワークで XOR を学習
- [ ] 勾配が正しく計算される
- [ ] 学習が収束する

## テストコード例

\`\`\`cpp
TEST(Phase2Integration, SimpleNeuralNetwork) {
    // データ: XOR 問題
    auto X = Tensor<float>({{0, 0}, {0, 1}, {1, 0}, {1, 1}});
    auto y = Tensor<float>({{0}, {1}, {1}, {0}});
    
    // パラメータ
    auto W1 = Variable<float>::randn({2, 4}, true);
    auto b1 = Variable<float>::zeros({4}, true);
    auto W2 = Variable<float>::randn({4, 1}, true);
    auto b2 = Variable<float>::zeros({1}, true);
    
    // Optimizer
    Adam<float> optimizer({&W1, &b1, &W2, &b2}, 0.01);
    
    // 学習
    for (int epoch = 0; epoch < 5000; ++epoch) {
        auto X_var = Variable<float>(X, false);
        auto h = relu(matmul(X_var, W1) + b1);
        auto y_pred = sigmoid(matmul(h, W2) + b2);
        auto loss = mse_loss(y_pred, Variable<float>(y, false));
        
        optimizer.zero_grad();
        loss.backward();
        optimizer.step();
    }
    
    // 精度確認
    // ...
}
\`\`\`

## 完了基準

- [ ] XOR が正しく解ける（精度 > 90%）
- [ ] メモリリークゼロ

## 関連ドキュメント

- [ROADMAP.md Phase 2 統合テスト](docs/ROADMAP.md)

## 期間

Week 6

## 依存タスク

- Depends on #12
